### PR TITLE
AEA-1180 More tests for documentation consistency (Continuation of #2081)

### LINKED ISSUE #2078: Missing tests for some docs sections
**Is your feature request related to a problem? Please describe.**
Some docs sections don't have tests

**Describe the solution you'd like**
Missing:
- [x] test that protobuf schema of envelope, message and protocols in docs (language-agnostic-definition.md) matches one we use; missing Dialogue Message etc
- [x] missing citations in and graphics for `agent-oriented-development.md`; missing discussion of time
- [x] vision section needs to define what an agent is
- [x] explain how to extend to further ledgers!
- [x] update Q&A
- [x] better explain lifecycle of the different packages in an AEA (use trade and skill guide example for this)
- [x] better explain search and discovery in skill guide
- [x] test all demos
- [x] update guides on storage...

### PR DESCRIPTION
**Pull request by marcofavorito**
## Proposed changes

Add more tests for documentation consistency with the actual code.

- [x] definining data models.md (proper python test)
- [x] generic storage (Same)
- [x] ledger integration (same)
- [x] multi-agent-manager (same)
- [x] oracle demo (bash)
- [x] query-language (python)
- [x] simple-oef-usage (python)
- [x] skill-testing (python)
- [x] quickstart has a test in it, we need to test the test

## Fixes

Fixes #2078 

## Types of changes

What types of changes does your code introduce to agents-aea?
_Put an `x` in the boxes that apply_

- [X] Bugfix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)

## Checklist

_Put an `x` in the boxes that apply._

- [X] I have read the [CONTRIBUTING](../CONTRIBUTING.md) doc
- [X] I am making a pull request against the `develop` branch (left side). Also you should start your branch off our `develop`.
- [ ] Lint and unit tests pass locally with my changes
- [ ] I have added tests that prove my fix is effective or that my feature works
- [ ] I have checked that code coverage does not decrease.
- [ ] I have checked that the documentation about the `aea cli` tool works
- [ ] I have added necessary documentation (if appropriate)
- [ ] Any dependent changes have been merged and published in downstream modules

## Further comments

n/a

### CODE REVIEW

File: tests/test_docs/test_bash_yaml/test_demo_docs.py [hunk 1]
```diff
@@ -33,25 +33,26 @@
...
-
-            yaml_code_blocks = extract_code_blocks(filepath=md_path, filter="yaml")
-            for blocks in yaml_code_blocks:
-                assert (
-                    blocks in bash_file
-                ), "[{}]: FAILED. Code must be identical".format(file)
-            logger.info("[{}]: PASSED".format(file))
+            # in the future, we might add other filters like "python"
+            self._test_blocks(file, "bash")
+            self._test_blocks(file, "yaml")
```
Comment (marcofavorito) [inline tests/test_docs/test_bash_yaml/test_demo_docs.py:58]: just a minor refactoring. I was planning to add `python` blocks, but I also think it's out of the scope of this PR.

File: aea/configurations/base.py [hunk 2]
```diff
@@ -1076,6 +1076,7 @@ class AgentConfig(PackageConfiguration):
     )
     CHECK_EXCLUDES = [
         ("private_key_paths",),
+        ("connection_private_key_paths",),
```
Comment (marcofavorito) [inline aea/configurations/base.py:1079]: @solarw @DavidMinarsch do we agree on this?

File: docs/generic-storage.md [hunk 3]
```diff
@@ -63,52 +63,49 @@ To get/put specific object collection instance should be used.
 Collection instance provide set of methods to handle data objects.
 List of collection methods:
 ```
-def put(self, object_id: str, object_body: JSON_TYPES) -> None:
-    """
-    Put object into collection.
+    def put(self, object_id: str, object_body: JSON_TYPES) -> None:
```
Comment (marcofavorito) [inline docs/generic-storage.md:66]: I added indentation so we can test consistency by string inclusion, without post-processing of the string.

File: docs/multi-agent-manager.md [hunk 4]
```diff
@@ -52,7 +60,7 @@ manager.add_agent(weather_station_id, agent_overrides=agent_overrides)
 component_overrides = {
     "name": "p2p_libp2p",
     "author": "fetchai",
-    "version": "0.9.0",
+    "version": "0.13.0",
     "type": "connection",
```
Comment (marcofavorito) [inline docs/multi-agent-manager.md:64]: @DavidMinarsch this wasn't detected by the bump script... we should update the script to detect such JSON objects.

**Comment by marcofavorito** [comment]
Overall it was a useful PR, as we detected some inconsistencies in documentation with the actual state of the framework. There is still room for improvement, as in most cases we only test that the code is "runnable", and not that the code works as it should.

**Review by DavidMinarsch — APPROVED** [review]
Excellent, thank you!

File: docs/multi-agent-manager.md (hunk 4 above)
Comment (DavidMinarsch) [reply docs/multi-agent-manager.md:64]: I think the bump script only locks at current vs next version. so it probably will be picked up going forward..

File: aea/configurations/base.py (hunk 2 above)
Comment (DavidMinarsch) [reply aea/configurations/base.py:1079]: correct

**Pull request merged**